src/transformers/processing_utils.py

```diff
@@ -993,6 +990,59 @@ def apply_chat_template(
         )
 
 
+def _check_reversed_images_text_for_vlms(images, text):
+    """
+    For backward compatibility: reverse the order of `images` and `text` inputs if they are swapped.
+    This method should only be called for processors where `images` and `text` have been swapped for uniformization purposes.
+    Note that this method assumes that two `None` inputs are valid inputs. If this is not the case, it should be handled
+    in the processor's `__call__` method before calling this method.
+    """
+
+    def _is_valid_text_input_for_processor(t):
+        if isinstance(t, str):
+            # Strings are fine
+            return True
+        elif isinstance(t, (list, tuple)):
+            # List are fine as long as they are...
+            if len(t) == 0:
+                # ... not empty
+                return False
+            elif isinstance(t[0], (str, int)):
+                # ... list of strings or int (for encoded inputs)
+                return True
+            elif isinstance(t[0], (list, tuple)):
+                # ... list of list of strings or int (for encoded inputs)
+                return isinstance(t[0][0], (str, int))
+            else:
+                return False
+        else:
+            return False
+
+    def _is_valid_or_convertible(input, validator, converter):
```

Comment:
nit - name implies it returns `True` if the input is either valid or convertible 

```suggestion
    def _is_valid_is_convertible(input, validator, converter):
```